doc/langref.html.in

```diff
@@ -302,6 +302,18 @@ const Timestamp = struct {
       in the middle of an expression, or just before a non-doc comment.
       </p>
       {#header_close#}
+      {#header_open|Top-Level Doc Comments#}
```

Comment:
I think we should call these container level doc comments and explain that they are meant to be used to document file structs and other anonymous containers.

Reply:
Do we explain anywhere in this document that files are structs? I don't want to lie to people, but I also don't want to over-complicate things.

Are there other anonymous structs for which doc comments make sense? Doc comments an non-pub things don't really go together.

Reply:
> Do we explain anywhere in this document that files are structs?

It is mentioned in the style guide but you don't have to specify it here.

> Are there other anonymous structs for which doc comments make sense? Doc comments and non-pub things don't really go together.

Anonymous structs are not necessarily non-pub, you could for example use them as struct field types.
```zig
pub const Foo = struct {
    bar: struct {
        //! docs here
        x: u32,
        y: u32,
    },
};
```
But I guess we could just require people to name their types if they want to document them.

Reply:
So do you still think I should change the name to "container-level" then? It's technically more correct, but I think your example is actually a good reason to keep it "top-level." The `bar` element in `Foo` really probably ought to be documented with a regular doc comment. For more complex anonymous structs, something like this seems better to me:

```zig
pub const Foo = struct {
    /// Docs for bar here - the type is less important than the
    /// specific use, since it's necessarily only ever used here.
    bar: struct {
        x: u32,
        y: u32,
        /// docs for specific elements of bar here
        pub fn baz() void {}
    },
};
```
Ultimately, it comes down to whether we want people to know how it works or how to use it. I think this document is better focused on the former.

Reply:
Well, I can't think of any situation where this would be needed at container level right now, so I guess I'm fine with either.